code/Form/UserFormsGridFieldFilterHeader.php

```diff
@@ -91,22 +92,28 @@ public function getHTMLFragments($gridField)
         )));
 
         foreach (array($start, $end) as $date) {
-            $date->setDateFormat('y-mm-dd');
+            $date->setDateFormat(DBDate::ISO_DATE);
             $date->addExtraClass('no-change-track');
         }
 
-        $end->setValue($state->end);
-        $start->setValue($state->start);
+        if ($state->end) {
+            $end->setValue($state->end);
+        }
+        if ($state->start) {
+            $start->setValue($state->start);
+        }
 
 
         $fields->push($actions = FieldGroup::create(
             GridField_FormAction::create($gridField, 'filter', false, 'filter', null)
-                ->addExtraClass('ss-gridfield-button-filter')
-                ->setAttribute('title', _t('SilverStripe\\Forms\\GridField\\GridField.Filter', "Filter"))
+                ->addExtraClass('ss-gridfield-button-filter btn btn-primary')
+                ->setTitle(_t(__CLASS__.'.FILTER', "Filter"))
+                ->setAttribute('title', _t(__CLASS__.'.FILTER', "Filter"))
                 ->setAttribute('id', 'action_filter_' . $gridField->getModelClass() . '_' . $columnField),
             GridField_FormAction::create($gridField, 'reset', false, 'reset', null)
-                ->addExtraClass('ss-gridfield-button-close')
-                ->setAttribute('title', _t('SilverStripe\\Forms\\GridField\\GridField.ResetFilter', "Reset"))
+                ->addExtraClass('ss-gridfield-button-close btn btn-primary')
```

Comment:
Do you have a screenshot of these class changes in action? I don't think we'd want two primary actions

Reply:
Hello @robbieaverill  yup here is a screenshot. I did some further modifications to set the colspan issue the filters were having, so it wont display an empty table cell on the right hand side. 
Here is a screenshot. 


![screen shot 2018-06-05 at 11 02 13 am](https://user-images.githubusercontent.com/143939/40956880-1cdca644-68b0-11e8-8ff7-81ccf7ea0926.png)

Reply:
Ok, thank you. Could you please fix the linting failures, then we're probably good to merge. https://travis-ci.org/silverstripe/silverstripe-userforms/jobs/388119897#L668

Reply:
Yup and thats fixed as well now